### PR TITLE
Sort by type_id in the order given in the &t= query param

### LINKED ISSUE #14: List (multiple) selected types at top of results
Sorting works properly with a single selected type:

http://localhost:3100/api/0.2/locations.json?api_key=EEQRBBUB&locale=en&muni=1&nelat=40.00156699355835&nelng=-105.27841178077733&swlat=40.00084067298544&swlng=-105.28098670143163&t=198

But fails to sort results properly for multiple selected types:

http://localhost:3100/api/0.2/locations.json?api_key=EEQRBBUB&locale=en&muni=1&nelat=40.00156699355835&nelng=-105.27841178077733&swlat=40.00084067298544&swlng=-105.28098670143163&t=198,136

### PR DESCRIPTION
**Pull request by dbalatero**
@ezwelty Closes #14

### CODE REVIEW

**Comment by ezwelty** [comment]
Interesting to have it sorted based on the order given in &t=. However, I believe the simpler binary sort is preferred, since it reflects how type filtering currently works in our apps: the user toggles types on/off, rather than sorting them in order of preference. In list view, we want to show all locations, sorted by distance, that fit the type filter criteria, and in map view (in places where there are more locations than can be loaded), we want to make sure that nearby locations that fit the type filter criteria are drawn preferentially.

So in conclusion, please return to a binary in/out sort. Also, to avoid making unnecessary changes to the API, please have locations with sort: 0 at the top of the list, followed by sort: 1, etc. As you can see, current behavior is to assign sort: 0 to locations that meet the type filter criteria:

https://fallingfruit.org/api/0.2/locations.json?api_key=EEQRBBUB&locale=en&muni=1&nelat=40.00156699355835&nelng=-105.27841178077733&swlat=40.00084067298544&swlng=-105.28098670143163&t=198

**Comment by dbalatero** [comment]
@ezwelty Got it, I think I must have misunderstood the issue. I can revert this and do the binary sort.
